### DISCUSSION TITLE
Receive Reject with Text="invalid integral value: " only when sending an Order Cancel/Repalce Request

### DISCUSSION THREAD

**Discussion by enricoZini** [quickfix-j/quickfixj#855] [Q&A] [answered]
When I try to replace a field with a decimal value, I receive a Reject message with this text: "invalid integral value: 0.8", with SessionRejectReason=INCORRECT_DATA_FORMAT_FOR_VALUE.

In the fix dictionary the field type is PERCENTAGE, which should accept decimal values.

I tested with other fields that also are of type PERCENTAGE and there was no problem.

When i send a NewOrderSingle with a decimal value in that field, there is no problem, only when I try to send a replace.

That text is found in the IntConverter class in the convert method, but i don't know why is it being called only when i send a Order Cancel/Replace Request.

I'm setting the field like this: "replace.setString(30022, textValue);", I tried using setDouble and setDecimal and had the same problem.

What can I do to prevent that field form being converted to an integer?

Thank you.

**Comment by the-thing** [comment]
"invalid integral value" is strictly associated with `quickfix.field.converter.IntConverter` which means that probably your field is not a decimal / percentage as you expect. I would start looking around that area.

It will be easier to help if you provide a unit test or minimal working example somewhere in your branch/fork.

**Reply by enricoZini** [reply]
Thank you for the answer.

I used the DataDictionary to get the field type and it returned that it is percentage.
dict.getFieldType(30022) -> FieldType[PERCENTAGE,class java.lang.Double,25]
Is there a better way to check the field type?

Also I think there would be the same problem when sending a NewOrderSingle, but it only occurs when sending a replace.

Maybe I should look into how the replace message type is handled.

**Reply by the-thing** [reply, marked as answer]
Messages can be created without a dictionary.

Not sure if you are populating messages using common messages class `quickfix.Message` or using a specific object such as `quickfix.fix44.OrderCancelReplaceRequest`.

30022 - I assume this is some custom field that you've created and probably it is not being set up correctly somewhere.

Make sure that either `Message` uses valid dictionary or FIX domain objects were generated using appropriate dictionary (if you are using custom ones).

**Reply by enricoZini** [reply]
It was a problem with the conterparty, where it was trying to convert that field to an integer only in the OrderCancelRepalceRequest, and then sending a Reject message

Sorry if i wasn't very clear when first asking the question.

**Reply by the-thing** [reply]
It wasn't clear, but I'm glad you sorted it out.
